src/network/networkd-link.c

```diff
@@ -1972,8 +1982,14 @@ static int link_joined(Link *link) {
                 r = link_set_bridge(link);
                 if (r < 0)
                         log_link_error_errno(link, r, "Could not set bridge message: %m");
+
         }
 
+        /* TODO: check for bridge interface? */
```

Comment:
yeah, tracking the link type in the Link object sounds like the much better idea. After all we should be compatible with people creating bridge interfaces with foreign tools, that networkd then takes possession of.

What's the best way to detect whether an interface is a bridge? where is that accessible via netlink?